src/Symfony/Component/Translation/Extractor/Visitor/ConstraintVisitor.php

```diff
@@ -0,0 +1,194 @@
+        'Locale',
+        'Luhn',
+        'Negative',
+        'NegativeOrZero',
+        'NotBlank',
+        'NotCompromisedPassword',
+        'NotEqualTo',
+        'NotIdenticalTo',
+        'NotNull',
+        'NumberConstraintTrait',
+        'Optional',
+        'Positive',
+        'PositiveOrZero',
+        'Range',
+        'Regex',
+        'Required',
+        'Time',
+        'Timezone',
+        'Traverse',
+        'Type',
+        'Unique',
+        'Url',
+        'Uuid',
+        'Valid',
+    ];
+
+    public function beforeTraverse(array $nodes): ?Node
+    {
+        return null;
+    }
+
+    public function enterNode(Node $node): ?Node
+    {
+        if (!$node instanceof Node\Expr\New_ && !$node instanceof Node\Attribute) {
+            return null;
+        }
+
+        $className = $node instanceof Node\Attribute ? $node->name : $node->class;
+        if (!$className instanceof Node\Name) {
+            return null;
+        }
+
+        $parts = $className->parts;
+        $isConstraintClass = false;
+
+        foreach ($parts as $part) {
+            if (\in_array($part, self::CONSTRAINT_CLASS_NAMES, true)) {
+                $isConstraintClass = true;
+
+                break;
+            }
+        }
+
+        if (!$isConstraintClass) {
+            return null;
+        }
+
+        if ($node instanceof Node\Expr\New_) {
+            return $this->extractFromConstraintClassInstantiation($node);
+        }
+
+        if ($node instanceof Node\Attribute) {
+            return $this->extractFromConstraintAttribute($node);
+        }
+
+        return null;
+    }
+
+    public function leaveNode(Node $node): ?Node
+    {
+        return null;
+    }
+
+    public function afterTraverse(array $nodes): ?Node
+    {
+        return null;
+    }
+
+    private function extractFromConstraintClassInstantiation(Node\Expr\New_ $node): ?Node
+    {
+        $arg = $node->args[0] ?? null;
+        if (!$arg instanceof Node\Arg) {
+            return null;
+        }
+
+        $options = $arg->value;
+        if (!$options instanceof Node\Expr\Array_) {
+            return null;
+        }
+
+        $message = null;
+        $messageNode = null;
+
+        foreach ($options->items as $item) {
+            if (!$item->key instanceof Node\Scalar\String_) {
+                continue;
+            }
+
+            if (false === stripos($item->key->value, 'message')) {
+                continue;
+            }
+
+            if (!$item->value instanceof Node\Scalar\String_) {
+                continue;
+            }
+
+            $message = $item->value->value;
+            $messageNode = $item;
+
+            break;
+        }
+
+        if (null !== $message && null !== $messageNode) {
+            $this->addMessageToCatalogue($message, 'validators', $node->getStartLine());
+        }
+
+        return null;
+    }
+
+    private function extractFromConstraintAttribute(Node\Attribute $node): ?Node
```

Comment:
let's use `void` as return type since we never return anything useful? or inline the logic altogether?
(same for the previous method)